packages/typespec-go/src/tcgcadapter/adapter.ts

```diff
@@ -15,14 +15,14 @@ const headerText = `Copyright (c) Microsoft Corporation. All rights reserved.
 Licensed under the MIT License. See License.txt in the project root for license information.
 Code generated by Microsoft (R) Go Code Generator.`;
 
-export function tcgcToGoCodeModel(context: EmitContext<GoEmitterOptions>): go.CodeModel {
+export async function tcgcToGoCodeModel(context: EmitContext<GoEmitterOptions>): Promise<go.CodeModel> {
   const info = new go.Info('TODO Title');
   const options = new go.Options(headerText, context.options['generate-fakes'] === true, context.options['inject-spans'] === true, context.options['disallow-unknown-fields'] === true);
   if (context.options['azcore-version']) {
     options.azcoreVersion = context.options['azcore-version'];
   }
 
-  const sdkContext = tcgc.createSdkContext(context);
+  const sdkContext = await tcgc.createSdkContext(context);
```

Comment:
Why was this function made async?

Reply:
this is from the latest breaking change of tcgc 0.44.2

Reply:
move to this pr: https://github.com/Azure/autorest.go/pull/1411